modules/gapi/test/common/gapi_core_tests.hpp

```diff
@@ -99,56 +108,197 @@ struct PrintNormCoreParams
-struct ResizeTestFxFy    : public TestWithParam<std::tuple<compare_f, int, int, cv::Size, double, double, cv::GCompileArgs>> {};
-struct Merge3Test        : public TestParams<std::tuple<cv::Size, cv::GCompileArgs>> {};
-struct Merge4Test        : public TestParams<std::tuple<cv::Size, cv::GCompileArgs>> {};
-struct RemapTest         : public TestParams<std::tuple<int,cv::Size,bool, cv::GCompileArgs>> {};
-struct FlipTest          : public TestParams<std::tuple<int, int, cv::Size,bool, cv::GCompileArgs>> {};
-struct CropTest          : public TestParams<std::tuple<int,cv::Rect,cv::Size,bool, cv::GCompileArgs>> {};
-struct ConcatHorTest     : public TestWithParam<std::tuple<int, cv::Size, cv::GCompileArgs>> {};
-struct ConcatVertTest    : public TestWithParam<std::tuple<int, cv::Size, cv::GCompileArgs>> {};
-struct ConcatVertVecTest : public TestWithParam<std::tuple<int, cv::Size, cv::GCompileArgs>> {};
-struct ConcatHorVecTest  : public TestWithParam<std::tuple<int, cv::Size, cv::GCompileArgs>> {};
-struct LUTTest           : public TestParams<std::tuple<int, int, cv::Size,bool, cv::GCompileArgs>> {};
-struct ConvertToTest     : public TestParams<std::tuple<int, int, cv::Size, double, double, compare_f, cv::GCompileArgs>> {};
-struct PhaseTest         : public TestParams<std::tuple<int, cv::Size, bool, cv::GCompileArgs>> {};
-struct SqrtTest          : public TestParams<std::tuple<int, cv::Size, cv::GCompileArgs>> {};
-struct NormalizeTest : public TestParams<std::tuple<compare_f,MatType,cv::Size,double,double,int,MatType,bool,cv::GCompileArgs>> {};
+struct MathOpTest        : public TestWithParamBase<mathOp,bool,double,bool>
+{
+    DEFINE_SPECIFIC_PARAMS_4(opType, testWithScalar, scale, doReverseOp);
+    USE_UNIFORM_INIT(MathOpTest);
+};
+struct MulDoubleTest     : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(MulDoubleTest);
+};
+struct DivTest           : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(DivTest);
+};
+struct DivCTest          : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(DivCTest);
+};
+struct MeanTest          : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(MeanTest);
+};
+struct MaskTest          : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(MaskTest);
+};
+struct Polar2CartTest    : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(Polar2CartTest);
+};
+struct Cart2PolarTest    : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(Cart2PolarTest);
+};
+struct CmpTest           : public TestWithParamBase<CmpTypes,bool>
+{
+    DEFINE_SPECIFIC_PARAMS_2(opType, testWithScalar);
+    USE_UNIFORM_INIT(CmpTest);
+};
+struct BitwiseTest       : public TestWithParamBase<bitwiseOp>
+{
+    DEFINE_SPECIFIC_PARAMS_1(opType);
+    USE_UNIFORM_INIT(BitwiseTest);
+};
+struct NotTest           : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(NotTest);
+};
+struct SelectTest        : public TestWithParamBase<>
+{
+    USE_UNIFORM_INIT(SelectTest);
+};
```

Comment:
I believe test definitions like this may be one-liners:
```c++
struct SelectTest        : public TestWithParamBase<> { USE_UNIFORM_INIT(SelectTest); };
```
So you get 4x less SLOC for the majority of tests. Tests with multiple magic spells inside may remain multi-line/

BTW why do you need to pass test name to the macro? it looks redundant.

Reply:
Done